src/analytics/AppAnalytics.ts

```diff
@@ -101,60 +101,61 @@ class AppAnalytics {
+          debug: __DEV__,
+          trackAppLifecycleEvents: true,
+          trackDeepLinks: true,
+          writeKey: SEGMENT_API_KEY,
+          storePersistor: AsyncStoragePersistor,
+        })
 
-      this.segmentClient.add({ plugin: new DestinationFiltersPlugin() })
-      this.segmentClient.add({ plugin: new InjectTraits() })
-      this.segmentClient.add({ plugin: new AdjustPlugin() })
-      this.segmentClient.add({ plugin: new ClevertapPlugin() })
-      if (FIREBASE_ENABLED) {
-        this.segmentClient.add({ plugin: new FirebasePlugin() })
-      }
+        this.segmentClient.add({ plugin: new DestinationFiltersPlugin() })
+        this.segmentClient.add({ plugin: new InjectTraits() })
+        this.segmentClient.add({ plugin: new AdjustPlugin() })
+        this.segmentClient.add({ plugin: new ClevertapPlugin() })
+        if (FIREBASE_ENABLED) {
+          this.segmentClient.add({ plugin: new FirebasePlugin() })
+        }
 
-      try {
-        const deviceInfo = await getDeviceInfo()
-        this.deviceInfo = deviceInfo
-        uniqueID = deviceInfo.UniqueID
-        this.sessionId = sha256(Buffer.from(uniqueID + String(Date.now()))).slice(2)
-      } catch (error) {
-        Logger.error(TAG, 'getDeviceInfo error', error)
-      }
+        try {
+          const deviceInfo = await getDeviceInfo()
+          this.deviceInfo = deviceInfo
+          uniqueID = deviceInfo.UniqueID
+          this.sessionId = sha256(Buffer.from(uniqueID + String(Date.now()))).slice(2)
+        } catch (error) {
+          Logger.error(TAG, 'getDeviceInfo error', error)
+        }
 
-      Logger.info(TAG, 'Segment Analytics Integration initialized!')
-    } catch (err) {
-      const error = ensureError(err)
-      Logger.error(TAG, `Segment setup error: ${error.message}\n`, error)
+        Logger.info(TAG, 'Segment Analytics Integration initialized!')
+      } catch (err) {
+        const error = ensureError(err)
+        Logger.error(TAG, `Segment setup error: ${error.message}\n`, error)
+      }
+    } else {
+      Logger.info(TAG, 'Segment API key not present, skipping setup')
     }
 
-    try {
-      const statsigUser = getDefaultStatsigUser()
-      // getAnonymousId causes the e2e tests to fail
-      let overrideStableID: string = E2E_TEST_STATSIG_ID
-      if (!isE2EEnv) {
+    if (STATSIG_ENABLED) {
+      try {
+        const statsigUser = getDefaultStatsigUser()
         if (!this.segmentClient) {
           throw new Error('segmentClient is undefined, cannot get anonymous ID')
         }
-        overrideStableID = this.segmentClient.userInfo.get().anonymousId
+        const overrideStableID = this.segmentClient.userInfo.get().anonymousId
+        Logger.debug(TAG, 'Statsig stable ID', overrideStableID)
+        await Statsig.initialize(STATSIG_API_KEY, statsigUser, {
+          // StableID should match Segment anonymousId
+          overrideStableID,
+          environment: STATSIG_ENV,
+          localMode: isE2EEnv,
```

Comment:
Should we just initialize the statsig client in local mode if statsig is disabled? That way we can avoid the custom gate overrides code too. And we can avoid the noisy logs by tweaking the condition. 

Also, it seems like for statsig to be enabled, we'll need the segment key too because of the stable id dependency. Wonder if we should remove that dependency. We can probably let statsig set it's own anonymous id in cases where segment is not enabled.

Reply:
ah I didn't know about the `localMode` option.
However my intention is to be able to run even if the statsig lib is not installed. So the approach here of not instantiating the client is preferrable. We're planning to make 3rd party deps like segment, clevertap, etc [optional in the modular runtime](https://linear.app/valora/issue/RET-1284/make-non-essential-3rd-party-dependencies-optional).

Reply:
> Also, it seems like for statsig to be enabled, we'll need the segment key too because of the stable id dependency. Wonder if we should remove that dependency. We can probably let statsig set it's own anonymous id in cases where segment is not enabled.

Yes I thought about that, and was on the fence whether to make that change now.
I'm not sure if we'll need statsig enabled without segment.
To run experiments we need segment to be connected to statsig, right?

Reply:
makes sense re running without installing those libraries. 
For experiments, yes segment is needed but it doesn't need to be. Metrics can potentially go to statsig directly without segment. Also, maybe there's a case where statsig is just needed for dynamic config. But I guess we could do that separately.

should we just drop the localMode option now that it is always going to be false?

Reply:
yes definitely I'll remove it.